File: filterls/filter.clj
#!/usr/bin/env clj

(require '[clojure.string :as str])

(defn filterer [line]
  (try 
    (> (Integer/parseInt (str/trim (get (str/split line #" " 3) 1))) 10)
    (catch Exception e false)))

(defn process [in out]
  (with-open 
    [
      bufout (java.io.BufferedWriter. out)
      bufin (java.io.BufferedReader. in)
    ]
    (doseq
      [l (line-seq bufin)]
      (if (filterer l) (do (.write bufout l) (.newLine bufout))))))

(defn process-file [fin fout]
  (process (java.io.FileReader. fin) (java.io.FileWriter. fout)))

(defn process-stdio [] (process *in* *out*))


(time (process-file "test-files/data.txt" "output/clojure.out"))

;(process (java.io.StringReader. (slurp "test-files/data.txt")))

File: filterls/filter.kt
fun isValid(s: String): Boolean {
    val segments = s.split(" ", limit = 3)
    if (segments.size < 2) {
        return false
    }
    return segments[1].toIntOrNull()?.let { it > 10 } ?: false
}

fun main(argv: Array<String>) {
    val out = java.io.BufferedWriter(java.io.PrintWriter(System.`out`))

    if (argv.isNotEmpty()) {
        java.io.File(argv[0]).forEachLine {
            if (isValid(it)) { out.write(it); out.newLine() }
        }
    } else {
        //while (true) {
        //    val line = readLine() ?:return
        //    if (isValid(line)) { println(line) }
        //}
        val bufin = java.io.BufferedReader(java.io.InputStreamReader(System.`in`))
        while (true) {
            val line = bufin.readLine() ?: break
            if (isValid(line)) { out.write(line); out.newLine() }
        }
    }

    out.flush()
}
